.github/workflows/codeql.yml

```diff
@@ -16,32 +16,45 @@
 name: CodeQL
 on:
   push:
-    branches: [ main ]
+    branches: [main]
   pull_request:
     # The branches below must be a subset of the branches above
-    branches: [ main ]
+    branches: [main]
 
 permissions:
   contents: read
   security-events: write
 
 jobs:
+  go_version:
+    runs-on: ubuntu-latest
```

Comment:
i dont think we need a exta job just to extract the version we can do in the same job

Reply:
Sure we can in-line it. The only reason I get it separate was that I was hoping it would be throwaway code that could eventually be removed once setup-go handles all of this better :)

Reply:
thanks

Reply:
ah wait, I remember why I made it a separate task now — chicken before the egg

I needed Go installed in order to use `go list -m -f 'version={{.GoVersion}}'`, but I can switch to just using awk and doing `awk -F'[ .]' '/^go /{print $2 "." $3; exit}' go.mod`

Reply:
Updated to in-line the steps into the existing job and to use awk